src/components/dialog/partial-styles/show-hide-animations.scss

```diff
@@ -0,0 +1,48 @@
+dialog {
+    --limel-dialog-display-speed: 0.2s;
+    transition:
+        opacity 0.2s ease,
+        display var(--limel-dialog-display-speed) ease,
+        transform var(--limel-dialog-display-speed) ease;
+    transition-behavior: allow-discrete;
+    opacity: 0;
+    transform: translate3d(0, 1rem, 0);
+
+    &::backdrop {
+        transition: background-color 0.6s ease;
+        transition-behavior: allow-discrete;
+        background-color: rgb(var(--color-black), 0);
+    }
+
+    &[open] {
+        --limel-dialog-display-speed: 0.4s;
+        opacity: 1;
+        transform: translate3d(0, 0, 0) scale(1);
+
+        @starting-style {
```

Comment:
SonarCloud is complaining about `transition-behavior` and `@starting-style`, it seems they are not implemented in Firefox yet. Will it work acceptable in Firefox anyway?

Reply:
no it's still [not fully supported yet](https://caniuse.com/?search=starting-style), but the transition will be fine. we basically don't get any transition at all in firefox at the moment. But it's ok I'd say. The alternative would be dynamically adding and removing classes, when the portal closes. Like what we did for the snackbars. But I'm a bigger fan of the browsers' builtin features, not such scripts that try to compensate for them. I think we can let this go. What do you say?

Reply:
It's fine, I just wanted to make sure you were aware :smile: 
I can silence the warnings from SonarCloud